### PR TITLE
Add availability data for mids not associated with a delivery service

### LINKED ISSUE #5074: CreateStats not adding availability data for <server>: not found in DeliveryServices
## I'm submitting a ...

-  bug report

## Traffic Control components affected ...

-  Traffic Monitor

## Current behavior:

These messages show up in the Traffic Monitor logs:

```
trafficmonitor_1            | INFO: stat.go:54: 2020-09-21T16:08:45.316647589Z: CreateStats not adding availability data for 'mid-01': not found in DeliveryServices
trafficmonitor_1            | INFO: stat.go:54: 2020-09-21T16:08:45.31670463Z: CreateStats not adding availability data for 'mid-02': not found in DeliveryServices
```

## Expected behavior:

Not having those warnings. This check can just be removed for mids, since topologies do not give information about mids in the snapshot.

## Minimal reproduction of the problem with instructions:

1. Make a topology with at least 1 mid cachegroup
2. Make a topology-based delivery service with the topology you made
3. Snapshot
2. Look at Traffic Monitor

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5074 OR is not related to any Issue 
- Skips the check for whether a cache is in a delivery service for mids, since the CrConfig does not contain that information about mids for topology-based delivery services

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Monitor

## What is the best way to verify this PR?

Run the TM unit tests

## If this is a bug fix, what versions of Traffic Control are affected?

- master (13cdcaa411)

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_monitor/ds/stat.go [hunk 1]
```diff
@@ -49,16 +49,16 @@ func addAvailableData(dsStats *dsdata.Stats, crStates tc.CRStates, serverCachegr
...
-			log.Infof("CreateStats not adding availability data for '%s': not found in DeliveryServices\n", cache)
-			continue
-		}
 		cacheType, ok := serverTypes[cache]
 		if !ok {
 			log.Infof("CreateStats not adding availability data for '%s': not found in Server Types\n", cache)
 			continue
 		}
+		deliveryServices, ok := serverDs[cache]
+		if !ok && cacheType != tc.CacheTypeMid {
```
Comment (ocket8888) [inline traffic_monitor/ds/stat.go:58]: This isn't sufficient to determine that a cache is not a mid. The approach varies from place to place - TP in some places checks for names matching `.*MID.*`, queries in TO typically do something like 'MID%' - but I think the most common is to check if the type name begins with `tc.MidTypePrefix`.

Comment (zrhoffman) [reply traffic_monitor/ds/stat.go:58]: The cache types are originally from `todata.getServerTypes()` which already converts the server type to a well-defined one:

```go
		t := tc.CacheTypeFromString(serverData.Type)
```

Also, `cacheType` is of type `tc.CacheType`, so the only possible values are `CacheType("EDGE")`, `CacheType("MID")`, and `CacheType("")`

Comment (ocket8888) [reply traffic_monitor/ds/stat.go:58]: Well thank god.

**Review by ocket8888 — APPROVED** [review]
Excessive warnings no longer appear, all tests passing

**Pull request merged**